concepts/basics/introduction.md

```diff
@@ -0,0 +1,83 @@
+# Introduction
+
+To start the C++ journey we are starting with variables, function calls, and comments.
+
+## Variables
+
+C++ is a typed language.
+The type of a variable has to be stated upon the declaration of a variable.
+Its type cannot change during its lifetime.
+An integer variable with the name `years` can be declared like this:
+
+```cpp
+int years;
+```
+
+It is good practice to initialize variables upon declaration.
```

Comment:
I think I should put that into the numbers concept. If you do not initialize a variable, it will be some kind of null-value. Which is bad form, but okay for an int type that will have "zero" for most compilers but might be garbage and cause unforeseen behaviour for reference types with uninitialized memory blocks.

Reply:
I added `This ensures that the content of the variable is well-defined and not filled with random data.`. Is this good reason enough?

Reply:
```exercism/caution
    C++ does allow using uninitialized variables.
    However, when you do the results can be unpredicable.
    [optionally offer examples: an undesirable situation, and a disastrous one]
    It is therefore good practice to initialize variables at their declaration whenever possible.
    That way it is impossible to forget to on time.
    ```

A disadvantage of «content» and «filled» and «random data» is that it might not be clear to a student what these should mean.

Reply:
Begs the question what «valid» means. Or what sorts of «invalid» data might even exist.

Coming from many other languages, the idea of uninitialized variables is foreign and _weird_.